src/com/amazon/ion/benchmark/DataConstructor.java

```diff
@@ -224,12 +228,78 @@ public static IonValue constructIonData(ReparsedType parsedTypeDefinition) {
                     return SYSTEM.newBlob(constructLobs(constraintMapClone));
                 case CLOB:
                     return SYSTEM.newClob(constructLobs(constraintMapClone));
+                case STRUCT:
+                    return constructIonStruct(constraintMapClone);
+                case LIST:
+                    return constructIonList(constraintMapClone);
                 default:
                     throw new IllegalStateException(type + " is not supported.");
             }
         }
     }
 
+    /**
+     * Constructing IonStruct which is aligned with the constraints provided in the constraintMap.
+     * @param constraintMapClone collects the constraints from ISL file, the key represents the name of constraints,
+     * and the value is constraint value in ReparsedConstraint format.
+     * @return the constructed IonStruct value.
+     */
+    private static IonValue constructIonStruct(Map<String, ReparsedConstraint> constraintMapClone) {
+        Fields fields = (Fields)constraintMapClone.remove("fields");
+        IonStruct constructedIonStruct = SYSTEM.newEmptyStruct();
+        Map<String, ReparsedType> fieldMap = fields.getFieldMap();
+        // Check if there is unhandled constraint provided.
+        if (!constraintMapClone.isEmpty()) {
+            throw new IllegalStateException ("Found unhandled constraints : " + constraintMapClone.values());
+        }
+        // Writing field value to IonStruct based on the relevant constraint.
+        for (String fieldName : fieldMap.keySet()) {
+            // Get the type definition for each field.
+            ReparsedType fieldTypeDefinition = fieldMap.get(fieldName);
```

Comment:
Since you use both the key and the value, it's slightly cleaner to iterate over `fieldMap.entrySet()`.

Reply:
> Since you use both the key and the value, it's slightly cleaner to iterate over `fieldMap.entrySet()`.

Agree, this change will be updated in the next commit.